packages/flutter/lib/src/widgets/will_pop_scope.dart

```diff
@@ -9,10 +9,87 @@ import 'routes.dart';
 /// Registers a callback to veto attempts by the user to dismiss the enclosing
 /// [ModalRoute].
 ///
+/// {@tool snippet --template=stateful_widget}
+///
+/// Whenever the back button is pressed, you will get a callback at [onWillPop],
+/// which returns a [Future]. If the [Future] returns true, the screen is
+/// popped.
+///
+/// ```dart
+/// bool shouldPop = true;
+/// @override
+/// Widget build(BuildContext) {
+///   return WillPopScope (
+///     onWillPop: () async {
+///       return shouldPop;
+///     },
+///     child: Center(
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// {@tool dartpad --template=stateful_widget_material}
+/// ```dart
+/// bool shouldPop = true;
+/// @override
+/// Widget build(BuildContext context) {
+///   return WillPopScope(
+///     onWillPop: () async {
+///       return shouldPop;
+///     },
+///     child: Scaffold(
+///       appBar: AppBar(
+///         title: Text("Flutter WillPopScope demo"),
+///       ),
+///       body: Center(
+///         child: Wrap(
```

Comment:
Column would be a better widget to use here. The long string of text is causing some layout issues:

<img width="374" alt="Screen Shot 2020-11-09 at 10 34 10 AM" src="https://user-images.githubusercontent.com/1145719/98582031-3cb6d600-2277-11eb-977b-fe55c0193ae6.png">


```
        body: Center(
          child: Column(
            children: [
              OutlinedButton(
```